packages/coil-extension/src/background/services/DisablingService.ts

```diff
@@ -0,0 +1,212 @@
+      this.tabStates.set(tab, { disabling })
+    }
+  }
+
+  /**
+   * Upon changes, must check all open tabs/frames to see if they apply and
+   * update the tab state accordingly.
+   *
+   * If ANY of the controls are set to disabled, disable ALL streams for the
+   * active tab, then apply the rules to other tabs (which may be backgrounded).
+   *
+   */
+  setDisabled(activeTab: number, controls: DisablingControls) {
+    console.log(
+      'setDisabled activeTab before:',
+      JSON.stringify(this.tabStates.get(activeTab), null, 2)
+    )
+    const priorState = this.getFrameStates()
+    this.tabStates.set(activeTab, { disabling: controls })
+
+    const topFrame = { tabId: activeTab, frameId: 0 }
+    const top = this.framesService.getFrameNotNull(topFrame)
+    const paymentPointer = this.tabStates.getFrameOrDefault(topFrame)
+      .monetizationDetails?.paymentPointer
+
+    const urlObject = new URL(top.href)
+    const { href, origin } = urlObject
+    this.disabled[href] = controls.disableUrl
+    this.disabled[origin] = controls.disableDomain
+    if (paymentPointer) {
+      this.disabled[paymentPointer] = controls.disablePaymentPointer
+    }
+
+    for (const state of priorState) {
+      const frame = state.frame
+      // Need to reapply
+      this.applyUrlBlocking(frame, state.href)
+      this.applyPaymentPointerBlocking(frame, state.paymentPointer)
+
+      const wasDisabled = state.disabled
+      const wasNotDisabled = !wasDisabled
+      const nowDisabled = this.checkIsFrameDisabled(frame)
+      const nowEnabled = !nowDisabled
+
+      // this.tabStates.setFrame(frame, { disabled: nowDisabled })
+      const tabState = this.tabStates.get(frame.tabId)
+      const { interactive } = this.tabStates.getFrameOrDefault(frame)
+      if (wasDisabled && nowEnabled) {
+        if (interactive || tabState.stickyState === 'sticky') {
+          if (tabState.playState === 'playing') {
+            this.resume(frame)
+          }
+        }
+      } else if (wasNotDisabled && nowDisabled) {
+        if (interactive || tabState.stickyState === 'sticky') {
+          if (tabState.playState === 'playing') {
+            this.pause(frame)
+          }
+        }
+      }
+    }
+
+    console.log(
+      'setDisabled activeTab after:',
+      JSON.stringify(this.tabStates.get(activeTab), null, 2)
+    )
+  }
+
+  getFrameStates() {
+    const ret: Array<{
+      frame: FrameSpec
+      disabled: boolean
+      href: string
+      paymentPointer: string | null
+    }> = []
+    for (const tabId of this.tabStates.tabKeys()) {
+      const tabState = this.tabStates.get(tabId)
+      for (const frameId of Object.keys(tabState.frameStates).map(Number)) {
+        const frame: FrameSpec = { frameId, tabId }
+        const state = tabState.frameStates[frameId]
+        if (!this.framesService.getFrame(frame)) {
+          console.log('getFrameStates missing frame', JSON.stringify(frame))
```

Comment:
TODO: not sure why we are getting this 
Potentially the tabStates frame bookkeeping isn't 100%
No ... 
It was due to stopWebMonetization deleting the frameState